src/main/java/io/github/jelastic/core/repository/ElasticRepository.java

```diff
@@ -320,6 +325,46 @@ public void reAlias(@NotEmpty  String newIndex, @NotEmpty String aliasName) {
         return ElasticUtils.getResponse(multiGetItemResponses, idSearchRequest.getKlass());
     }
 
+    /**
+     * Load data of entire index based on batchSize using ES scroll API
+     * @param <T> Response class Type
+     * @param index Index to be loaded
+     * @param query jelastic query object
+     * @param batchSize Will tell ElasticClient the size of each fetch, should be <= 10000 for better performance
+     * @return List<T> list of all objects in that index
+     */
+    public <T> List<T> loadAll(String index, Query query, Integer batchSize, Class<T> klass) {
+        final Scroll scroll = new Scroll(TimeValue.timeValueMinutes(1L));
+
+        QueryBuilder queryBuilder = queryManager.getQueryBuilder(query);
+        SearchRequestBuilder searchRequestBuilder = elasticClient
+                .getClient()
+                .prepareSearch(index)
+                .setQuery(queryBuilder)
+                .setSize(batchSize)
+                .setScroll(scroll);
+        SearchResponse searchResponse = searchRequestBuilder
+                .execute()
+                .actionGet();
+
+        String scrollId = searchResponse.getScrollId();
+        List<T> batchedResult = ElasticUtils.getResponse(searchResponse, klass);
+        List<T> totalResult = new ArrayList<>(batchedResult);
+        int count = 1;
+        while (!batchedResult.isEmpty()) {
```

Comment:
Couple of things. Please add the description on how you are fetching the scroll results in batches on top of the method. Second, you can reduce this to a function-pointer instead into a different private method and define your recursion better than a primitive way such as this!

Reply:
1. Description will be added.
2. Regarding using lambda for iteration, thing is I am not iterating **on** any iterable or stream. What I am doing is imitating infinite loop, unless all the data is read into the List. Once no data is returned by ES, the infinite loop terminates. Other way would be using `IntStream.iterate` to imitate infinite loop, I feel that would be overkill for such simple use case just for sake of readability .

Reply:
I agree. Intstream.iterate may be an overkill.